packages/flutter/lib/src/material/text_field.dart

```diff
@@ -387,10 +387,18 @@ class _TextFieldState extends State<TextField> {
       );
     }
 
-    return new GestureDetector(
-      behavior: HitTestBehavior.opaque,
-      onTap: _requestKeyboard,
-      child: child,
+    return new Semantics(
+      onTap: () {
+        _requestKeyboard();
```

Comment:
_requestKeyboard() is async, I believe. To make this code clearer, I recommend putting that line below the next two, that way it's more obvious what order things will happen in (and more obvious that the keyboard isn't relevant to the next two lines).